src/ripple/overlay/impl/Message.cpp

```diff
@@ -41,18 +42,53 @@ Message::Message (::google::protobuf::Message const& message, int type)
 
     mBuffer.resize (headerBytes + messageBytes);
 
-    auto ptr = mBuffer.data();
+    auto set_header = [](uint8_t *ptr, decltype(messageBytes) messageBytes, int type, bool compressed=false,
```

Comment:
Consider a type alias for `decltype(messageBytes)`. I don't love `decltype(messageBytes) messageBytes`, it mixes variables from different scopes.